### DISCUSSION TITLE
Is that possible to use api key/access token to use Airflow stable API?

### DISCUSSION THREAD

**Discussion by yuvanpradeep** [apache/airflow#26928] [Q&A]
I am currently using a basic authentication mechanism and it is working fine but wanted to know if we can use some other mechanism.

Scenario: Service will use the Airflow API to trigger/pool status of the dags
Airflow version: 2.1.4
Reference: https://airflow.apache.org/docs/apache-airflow/2.1.4/security/api.html

**Comment by potiuk** [comment]
Depends on your authentication backend. There are few authentication backends shipped with airflow  but you can roll your own authentication backend if you want: https://airflow.apache.org/docs/apache-airflow/stable/security/api.html?highlight=authentication+backend#roll-your-own-api-authentication

You can also use basic authentication as equivalent of - token authentication and generate randomly generated password for the user you create (using CLI for example) and it will be pretty much equivalent to using Token (simply the token is the randomly generated string that you set as password). In this case you just need to specify https://airflow.apache.org/docs/apache-airflow/stable/security/api.html?highlight=authentication+backend#basic-authentication  - this will work the same way as token  based authentication, you just need to implement the token generation mechanism if you want to expose it users to generate their own tokens.
